### DISCUSSION TITLE
To help with plugin/package dev, allow specifying namespace in artisan make commands?

### DISCUSSION THREAD

**Discussion by ChrisHardie** [filamentphp/filament#2040] [Ideas]
In building a custom package that includes Filament and provides Filament resources, I'm having to generate Filament resources in a test Laravel app and then copy those over from `app/Filament` to the package's `src/Filament` directory and then update all of the `namespace` and `use` statements accordingly.

I wonder, would it be useful to submit a PR to the Filament "make" commands to allow the user to specify a different namespace for Filament in the resulting generated files? Or is this a limited enough use case that it's not worth it?

**Comment by danharrin** [comment]
It's definitely a problem, but you'd still need an app to actually execute the commands though, right?

**Reply by ChrisHardie** [reply]
That's right. I've been playing with getting the [orchestra/testbench](https://github.com/orchestral/testbench) setup to handle it and in theory it could all be self-contained, but best practices for such things seem to still be emerging.

**Reply by ChrisHardie** [reply]
If anything, I could just submit a PR to expand the Filament plugin documentation to clarify how one might generate and add new resources to the plugin until a better process is in place.
